File: src/lib/ir/__fixtures__/isomorphic/bfs.cpp
#include <vector>

int bfs(std::vector<int> queue) {
    int visited = 0;
    while (!queue.empty()) {
        int node = queue.back();
        queue.pop_back();
        if (node >= 0) {
            visited++;
        }
    }
    return visited;
}

File: src/lib/ir/__fixtures__/isomorphic/bfs.java
import java.util.List;

class Bfs {
    int bfs(List<Integer> queue) {
        int visited = 0;
        while (!queue.isEmpty()) {
            int node = queue.remove(queue.size() - 1);
            if (node >= 0) {
                visited++;
            }
        }
        return visited;
    }
}
